shimmingtoolbox/dicom_to_nifti.py

```diff
@@ -287,3 +344,59 @@ def fix_tfl_b1(nii_b1, json_data):
 
     # Return a fixed NIfTI object with the corrected affine matrix and the reshuffled/rescaled B1+ maps
     return nib.Nifti1Image(b1_complex, affine, header=nii_b1.header)
+
+def fix_xfl_b1(xfl_dict):
```

Comment:
consider changing the name 'fix' for something else more specific. 'Fix' sounds like something is broken and this function is fixing it, whereas IIUC this function is doing channel combinations and scaling.